Don't use FreeBSD's native off_t for z_off_t, as it may be too large
for zlib.

diff -Naur zconf.h.in.orig zconf.h.in
--- zconf.h.in.orig	2024-03-26 15:50:09.710081766 -0400
+++ zconf.h.in	2024-03-26 15:56:27.703416769 -0400
@@ -509,6 +509,14 @@
 #  define SEEK_END        2       /* Set file pointer to EOF plus "offset" */
 #endif
 
+/*
+ * This is hard-configured for FreeBSD, since zlib doesn't actually support
+ * using the system off_t for offsets unless off_t is no longer than long.
+ * To minimize the diff, we just "undef z_off_t" rather than modifying
+ * the following lines.
+ */
+#undef z_off_t
+
 #ifndef z_off_t
 #  define z_off_t long
 #endif
